lib/generators/query/query_generator.rb

```diff
@@ -0,0 +1,51 @@
+class QueryGenerator < Rails::Generators::NamedBase
+
+  def create_query_object_file
+    create_file "app/queries/#{file_name}_query.rb", <<-FILE
+class #{class_name}Query < BaseQuery
+
+  class << self
```

Comment:
this is not needed in sub-classes if inheriting from BaseQuery is it? but feel free to merge